index.js

```diff
@@ -125,3 +128,57 @@ app.listen(port, function() {
     if (flags.get('dry-run'))
         console.log('Starting in DRY-RUN mode');
 });
+
+function pullRequestPuller() {
+    waitFor(5 * 1000).then(function() {
+        github.get("/repos/:owner/:repo/pulls", {}).then(function (pull_requests) {
+            pull_requests.forEach(function(pull_request) {
+		        if (!webkit.related(pull_request.title)) {
+                    return;
+                }
+                /*
+                   // TODO: Do we want skip the "do not merge yet" PR?
+                   if (filter.pullRequestByLabel(pull_request, "do not merge yet", logArgs)) {
+                    return;
+                } */
+                metadata(pull_request.number, pull_request.user.login, pull_request.title, pull_request.body).then(function(metadata) {
+                    // logArgs(metadata);
+                    console.log("Issue: " + metadata.issue);
+                    console.log("Title: " + metadata.title);
+                    console.log("Labels: " + metadata.labels);
+                    console.log("isWebKitVerified: " + metadata.isWebKitVerified);
+                    console.log("isMergeable: " + metadata.isMergeable);
+                    console.log("reviewedDownstream: " + metadata.reviewedDownstream);
+
+                    if (metadata.isWebKitVerified) {
+                        console.log("Webkit flags:");
+                        if (metadata.webkit.flags.reviewed) {
+                              console.log("  Reviewed");
+                        }
+                        if (metadata.webkit.flags.inCommit) {
+                              console.log("  inCommit");
+                        }
+                    }
+                    // TODO: This part of the code must be uncommmented to
+                    /* return labelModel.post(n, metadata.labels).then(
+                         funkLogMsg(n, "Added missing LABELS if any."),
+                         funkLogErr(n, "Something went wrong while adding missing LABELS.")
+                    ).then(function() {
+                        return comment(n, metadata);
+                    }).then(
+                        funkLogMsg(n, "Added missing REVIEWERS if any."),
+                        funkLogErr(n, "Something went wrong while adding missing REVIEWERS.")
+                    ); */
+                });
+			});
+            pullRequestPuller();
```

Comment:
I think it is safe. the pullRequestPuller()  relays in the an async_func/then/promise chain. 
The pullRequestPuller() functions simply doesn't wait for the async functions so reach its end and just the promises is added the callback chains because the then() hadlers:

   waitFor(60 * 1000).then() -> github.get().then()

Just when the callback is resolved the recursive pullRequestPuller() call is invoked but the previous one was gone time ago.

Reply:
in the top of that,  the memory. I also ran a profiling with the `npm --inspect` for 15 minutes in order to check memory allocations leaks (heap and stack). The graph remains looks good:

![Screenshot from 2019-12-19 14-13-28](https://user-images.githubusercontent.com/376838/71176352-f5dc7e00-2269-11ea-939a-78d5904a16f6.png)